openapi/openapi_spec_resource_schema_definition_property_test.go

```diff
@@ -333,26 +569,137 @@ func TestTerraformSchema(t *testing.T) {
 			})
-			Convey("And the array elements are of the default type string (only supported type for now)", func() {
+			Convey("And the array elements are of type string", func() {
 				So(reflect.TypeOf(tfPropSchema.Elem).Elem(), ShouldEqual, reflect.TypeOf(schema.Schema{}))
 				So(tfPropSchema.Elem.(*schema.Schema).Type, ShouldEqual, schema.TypeString)
 			})
 		})
 	})
 
+	Convey("Given a swagger schema definition that has a property of type 'array' and the elems are type integer", t, func() {
+		s := &specSchemaDefinitionProperty{
+			Name:           "array_prop",
+			Type:           typeList,
+			ArrayItemsType: typeInt,
+			ReadOnly:       false,
+			Required:       true,
+		}
+		Convey("When terraformSchema method is called", func() {
+			tfPropSchema, err := s.terraformSchema()
+			Convey("Then the resulted terraform property schema should be of type array too", func() {
+				So(err, ShouldBeNil)
+				So(tfPropSchema.Type, ShouldEqual, schema.TypeList)
+			})
+			Convey("And the array elements are of type int", func() {
+				So(reflect.TypeOf(tfPropSchema.Elem).Elem(), ShouldEqual, reflect.TypeOf(schema.Schema{}))
+				So(tfPropSchema.Elem.(*schema.Schema).Type, ShouldEqual, schema.TypeInt)
+			})
+		})
+	})
+
+	Convey("Given a swagger schema definition that has a property of type 'array' and the elems are type number", t, func() {
+		s := &specSchemaDefinitionProperty{
+			Name:           "array_prop",
+			Type:           typeList,
+			ArrayItemsType: typeFloat,
+			ReadOnly:       false,
+			Required:       true,
+		}
+		Convey("When terraformSchema method is called", func() {
+			tfPropSchema, err := s.terraformSchema()
+			Convey("Then the resulted terraform property schema should be of type array too", func() {
+				So(err, ShouldBeNil)
+				So(tfPropSchema.Type, ShouldEqual, schema.TypeList)
+			})
+			Convey("And the array elements are of type float", func() {
+				So(reflect.TypeOf(tfPropSchema.Elem).Elem(), ShouldEqual, reflect.TypeOf(schema.Schema{}))
+				So(tfPropSchema.Elem.(*schema.Schema).Type, ShouldEqual, schema.TypeFloat)
+			})
+		})
+	})
+
+	Convey("Given a swagger schema definition that has a property of type 'array' and the elems are type bool", t, func() {
+		s := &specSchemaDefinitionProperty{
+			Name:           "array_prop",
+			Type:           typeList,
+			ArrayItemsType: typeBool,
+			ReadOnly:       false,
+			Required:       true,
+		}
+		Convey("When terraformSchema method is called", func() {
+			tfPropSchema, err := s.terraformSchema()
+			Convey("Then the resulted terraform property schema should be of type array too", func() {
+				So(err, ShouldBeNil)
+				So(tfPropSchema.Type, ShouldEqual, schema.TypeList)
+			})
+			Convey("And the array elements are of type bool", func() {
+				So(reflect.TypeOf(tfPropSchema.Elem).Elem(), ShouldEqual, reflect.TypeOf(schema.Schema{}))
+				So(tfPropSchema.Elem.(*schema.Schema).Type, ShouldEqual, schema.TypeBool)
+			})
+		})
+	})
```

Comment:
Simple cases, loop?

Reply:
as discussed, tackling this in a separate changeset